lib/plug/cowboy.ex

```diff
@@ -125,6 +125,33 @@ defmodule Plug.Cowboy do
 
   To opt-out of this default instrumentation, you can manually configure
   cowboy with the option `stream_handlers: [:cowboy_stream_h]`.
+
+  ## WebSocket support
+
+  Plug.Cowboy supports upgrading HTTP requests to WebSocket connections via 
+  the use of the `Plug.Conn.upgrade_adapter/3` function, called with `:websocket` as the second
+  argument. Applications should validate that the connection represents a valid WebSocket request
+  before calling this function (Cowboy will validate the connection as part of the upgrade
+  process, but does not provide any capacity for an application to be notified if the upgrade is
+  not successful). If an application wishes to negotiate WebSocket subprotocols or otherwise set
+  any response headers, it should do so before calling `Plug.Conn.upgrade_adapter/3`.
+
+  The third argument to `Plug.Conn.upgrade_adapter/3` defines the details of how Plug.Cowboy
+  should handle the WebSocket connection, and must take the form `{handler, handler_opts,
+  connection_opts}`, where values are as follows:
+
+  * `handler` is a module which implements the `:cowboy_websocket` behaviour. Note that this
```

Comment:
Let's add a link to `cowboy_websocket`?

Reply:
/me looks up how to link to Erlang behaviours in ex_doc

Reply:
It has to be manual:

```suggestion
  * `handler` is a module which implements the [`:cowboy_websocket`](https://ninenines.eu/docs/en/cowboy/2.6/manual/cowboy_websocket/) behaviour. Note that this
```

Reply:
Updated